packages/experiment-tag/src/experiment.ts

```diff
@@ -51,14 +55,21 @@ export const initializeExperiment = (apiKey: string, initialFlags: string) => {
     user = {};
   }
 
-  // create new user if it does not exist, or it does not have device_id
-  if (Object.keys(user).length === 0 || !user.device_id) {
-    user = {};
-    user.device_id = UUID();
-    globalScope.localStorage.setItem(
-      experimentStorageName,
-      JSON.stringify(user),
-    );
+  // create new user if it does not exist, or it does not have device_id or web_exp_id
+  if (Object.keys(user).length === 0 || !user.device_id || !user.web_exp_id) {
+    if (!user.device_id || !user.web_exp_id) {
+      // if user has device_id, migrate it to web_exp_id
+      if (user.device_id) {
+        user.web_exp_id = user.device_id;
+      } else {
+        const uuid = UUID();
+        user = { device_id: uuid, web_exp_id: uuid };
```

Comment:
since we call `setUser` with this user, we will never use the device ID passed from the experiment integration (analytics SDK). I think we only want to store the `web_exp_id` in the user, right? Or am I missing something?